pallets/rc-migrator/src/lib.rs

```diff
@@ -28,6 +28,33 @@
 //! To simplify development and avoid the need to edit the original pallets, this pallet may
 //! duplicate private items such as storage entries from the original pallets. This ensures that the
 //! migration logic can be implemented without altering the original implementations.
+//!
+//! ### Roles:
+//! - **Admin**: Defined by [`Config::AdminOrigin`] (typically Root and Fellows origins).
+//!   - Has all permissions available to other roles.
+//!   - `set_manager` - allowed to set the manager account ID.
+//!
+//! - **Manager**: An optional role assigned to an account via the `set_manager` call.
+//!   - `schedule_migration` - able to schedule migration to start at a specific time.
+//!   - `force_set_stage` - permitted to forcefully set the migration stage.
+//!   - `set_canceller` - authorized to set the canceller account ID.
+//!   - `pause_migration` - may pause the migration.
+//!   - `receive_query_response` - may acknowledge XCM message sent to the Asset Hub.
+//!   - `set_unprocessed_msg_buffer` - controls the unprocessed message buffer size (i.e., the
+//!     number of unprocessed messages allowed before the RC migrator temporarily pauses sending
+//!     data to the Asset Hub).
+//!   - `set_ah_ump_queue_priority` - manages the priority of the Asset Hub UMP queue during
+//!     migration. The priority pattern determines how often the AH UMP queue is prioritized over
+//!     others during migration. See [`Config::AhUmpQueuePriorityPattern`] for details.
+//!   - `send_xcm_message` - can send XCM messages during migration to the Asset Hub as signed by
```

Comment:
this is quite powerful, what is the effective dispatch origin of the sent XCM message on AH? is it RC/root?

Reply:
https://github.com/polkadot-fellows/runtimes/blob/64fca747c46f511c9010652c1fb58bc360218b77/pallets/rc-migrator/src/lib.rs#L1200-L1202